### PR TITLE
shuffle dice

### PR DESCRIPTION
**Pull request by leannezhang**

### CODE REVIEW

File: src/components/Game.jsx [hunk 1]
```diff
@@ -38,19 +39,22 @@ class Game extends Component {
 
     const mergeWordList = Object.assign({}, this.state.wordList)
 
-    if (!mergeWordList[word]) {
+    if ( word.length > 0 && !mergeWordList[word] ) {
```
Comment (BennyLy604) [inline src/components/Game.jsx]: More than 0?  Isn't the minimum length 2 or 3?

File: src/components/Common/shuffleBoard.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+const facesNum = 6
+const boardLength = 5
+const randomlySelectFace = (faces) => {
```
Comment (BennyLy604) [inline src/components/Common/shuffleBoard.js]: randomlySelectedFace

File: src/components/Common/shuffleBoard.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  let randomIndex = Math.floor(Math.random() * facesNum)
+  let face = faces.charAt(randomIndex)
+  return face
+}
+
+
+const shuffleDices = (dices) => {
```
Comment (BennyLy604) [inline src/components/Common/shuffleBoard.js]: shuffleDice
singular is die, plural is dice.

Comment (leannezhang) [reply src/components/Common/shuffleBoard.js]: alright, English teacher!

File: src/components/Game.jsx (hunk 1 above)
Comment (leannezhang) [reply src/components/Game.jsx]: Yes, if it's blank, we don't want to add it to the list.  The requirement didnt say min length of 2. If length is 3, the point is 1. I assume that length between 1 and 3 has a score of 0.

File: src/components/Game.jsx [hunk 4]
```diff
@@ -7,17 +8,64 @@ class Game extends Component {
...
+  //    }
+  //}
+  handleClick(letter) {
+    this.setState({
+      currentWord: this.state.currentWord.concat(letter)
+    })
+  }
+
+  calculateScore(word) {
+    return word.length
```
Comment (BennyLy604) [inline src/components/Game.jsx:35]: Wasn't there more logic to the scoring?  It was max of 8 or something and word.length - 2 I think.

File: src/components/ScoreBox/ScoreBox.jsx [hunk 5]
```diff
@@ -1,21 +1,33 @@
...
+  const words = Object.keys(wordList)
+  const scores = Object.values(wordList)
+
   return (
     <div className="score-box">
       <div className="word-list">
         <div className="words">
           <li>WORD</li>
-          <li>crane</li>
+          {
```
Comment (BennyLy604) [inline src/components/ScoreBox/ScoreBox.jsx:13]: Nice use of lambdas.

File: src/components/ScoreBox/ScoreBox.jsx [hunk 6]
```diff
@@ -1,21 +1,33 @@
...
+            words.map(function(word) {
+             return (<li>{word}</li>)
```
Comment (BennyLy604) [inline src/components/ScoreBox/ScoreBox.jsx]: Indenting issue

File: src/components/Game.jsx (hunk 4 above)
Comment (leannezhang) [reply src/components/Game.jsx:35]: Yes. You are correct. Wait for next commit.

File: src/components/Common/shuffleBoard.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
+const facesNum = 6
+const boardLength = 5
+const randomlySelectFace = (faces) => {
+  let randomIndex = Math.floor(Math.random() * facesNum)
+  let face = faces.charAt(randomIndex)
+  return face
```
Comment (BennyLy604) [inline src/components/Common/shuffleBoard.js]: Just return faces.charAt(randomIndex)

File: src/components/ScoreBox/ScoreBox.jsx (hunk 6 above)
Comment (leannezhang) [reply src/components/ScoreBox/ScoreBox.jsx]: how?

File: src/components/ScoreBox/ScoreBox.jsx (hunk 5 above)
Comment (leannezhang) [reply src/components/ScoreBox/ScoreBox.jsx:13]: Thanks teacher!

File: src/components/Common/shuffleBoard.js [hunk 8]
```diff
@@ -41,14 +40,14 @@ export const shuffleBoard = () => {
     ['','','','','']
   ]
 
-  const shuffledDices = shuffleDices(dices);
+  const shuffleddice = shuffledice(dice);
```
Comment (BennyLy604) [inline src/components/Common/shuffleBoard.js]: Camel case plz

File: src/components/ScoreBox/ScoreBox.jsx (hunk 6 above)
Comment (BennyLy604) [reply src/components/ScoreBox/ScoreBox.jsx]: The other lines have 2 character indent.  This line has 1 character indent.

File: src/components/Board/Board.jsx [hunk 9]
```diff
@@ -26,14 +13,13 @@ const Board = (props) => {
               <div className="row" key={index}>
                 {
                   row.map((tile, index) => {
-                    return (<Tile selected={tile.isSelected} letter={tile.letter} key={index} />)
+                    return (<Tile selected={tile.isSelected} letter={tile} key={index} handleClick={handleClick.bind(this, tile)}/>)
                   })
                 }
-            </div>
+              </div>
```
Comment (BennyLy604) [inline src/components/Board/Board.jsx]: Hmm.. maybe change to 2 character indent to keep it all consistent.  Or use tabs.  Either way, consistency looks nicer.

Comment (leannezhang) [reply src/components/Board/Board.jsx]: done. it was 2 characters for each tab.

Comment (BennyLy604) [reply src/components/Board/Board.jsx]: This is still 4 spaces.

**Pull request merged**